initramfs/src/filesystem.rs

```diff
@@ -0,0 +1,68 @@
+use bytes::Bytes;
+use cpio::NewcBuilder;
+use flate2::read::GzDecoder;
+use libflate::gzip;
+use log::debug;
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{Cursor, Read},
+};
+use tar::Archive;
+
+pub struct Filesystem {
+    files: HashMap<String, (NewcBuilder, Cursor<Vec<u8>>)>,
+}
+
+impl Filesystem {
+    pub fn new() -> Self {
+        Self {
+            files: HashMap::new(),
+        }
+    }
+
+    pub fn parse_files(&mut self, archives: Vec<Archive<GzDecoder<Cursor<Bytes>>>>) {
+        // We need to reverse the layers so that the last layer is the first one in the cpio
+        for mut archive in archives.into_iter().rev() {
+            // Iterate over all the files in the layer
+            for entry in archive.entries().unwrap() {
+                let mut entry = entry.unwrap();
+                let path = entry.header().path().unwrap().to_str().unwrap().to_string();
```

Comment:
I think you should prevent using `unwrap` and just return an error

Reply:
I fixed this while force pushing for the last time